.pre-commit-config.yaml

```diff
@@ -50,11 +50,15 @@ repos:
         stages: [manual] # Not automatically triggered, invoked via `pre-commit run --hook-stage manual clang-tidy`
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.7.3
+    rev: v0.8.4
     hooks:
       - id: ruff
         args: [--fix]
+        files: (\.py|SConstruct|SCsub)$
+        types_or: [text]
       - id: ruff-format
+        files: (\.py|SConstruct|SCsub)$
+        types_or: [text]
```

Comment:
This is needed because pre-commit decides which files to run on *before* handling potential explicit includes in `pyproject.toml`, at least for me on Linux.

And since the default `types:` is only known Python file types, it doesn't include the SConstruct and SCsub files.